hathor/manager.py

```diff
@@ -441,13 +441,15 @@ def _initialize_components(self) -> None:
         # self.start_profiler()
         if self._full_verification:
             self.log.debug('reset all metadata')
-            for tx in self.tx_storage.get_all_transactions():
-                tx.reset_metadata()
+            with self.tx_storage.allow_partially_validated_context():
+                for tx in self.tx_storage.get_all_transactions():
+                    tx.reset_metadata()
 
         self.log.debug('load blocks and transactions')
         for tx in self.tx_storage._topological_sort_dfs():
```

Comment:
Same thing here: should it go through only transactions in the scope?

Reply:
That makes sense, I'm almost sure that it will be implicit as long as `get_all_transaction` respects the `tx_storage.allow_scope`.

Reply:
Great! Thanks!